src/TRestRawSignal.cxx

```diff
@@ -716,9 +717,9 @@ void TRestRawSignal::Print() {
 /// \brief It builds a TGraph object that can be used for drawing.
 ///
 TGraph* TRestRawSignal::GetGraph(Int_t color) {
-    if (fGraph != NULL) {
+    if (fGraph) {
```

Comment:
imo abusing implicit conversion from a type T to bool is bad practice. This notation should only be used for actual `bool` variables.

Reply:
I am not sure if I agree with you here, for example in https://stackoverflow.com/questions/3825668/checking-for-null-pointer-in-c-c many people prefer this approach. I guess its a matter of preference but in my opinion this is more concise and thus better, I can't image how this could be a problem, even smart pointers support this notation.

Reply:
In terms of "correctness" if you will, this is perfectly fine of course. But if you consider programming from a standpoint of being an asynchronous discussion between multiple people reading the code, the concise approach is unclear to the reader that isn't perfectly aware of the full picture. There's no way to know that `fGraph` here is some pointer. It very well may be a flag that decides whether some graphs are created! 

In any more strongly typed language you'd have to type this out fully as well. And for good reason.